File: application/src/main/java/org/sheedon/common/handler/ToastHandler.java
package org.sheedon.common.handler;

import android.app.Application;
import android.os.Handler;
import android.os.Looper;
import android.text.TextUtils;

import androidx.annotation.StringRes;

import org.sheedon.common.app.BaseApplication;
import org.sheedon.common.utils.ToastUtils;

/**
 * 消息提示者
 *
 * @Author: sheedon
 * @Email: sheedonsun@163.com
 * @Date: 2021/10/19 6:04 下午
 */
public class ToastHandler {

    private static final ToastHandler INSTANCE = new ToastHandler();
    // ui 处理器
    private final Handler uiHandler = BaseApplication.getUiHandler();

    private ToastHandler() {
    }

    /**
     * 显示一个Toast
     *
     * @param msg 字符串
     */
    public static void showToast(final String msg) {
        if (TextUtils.isEmpty(msg)) {
            return;
        }
        if (Looper.myLooper() != INSTANCE.uiHandler.getLooper()) {
            INSTANCE.uiHandler.post(() -> ToastUtils.showToast(BaseApplication.getInstance(), msg));
        }
    }

    /**
     * 显示一个Toast
     *
     * @param msgId 传递字符串的资源
     */
    public static void showToast(@StringRes int msgId) {
        Application application = BaseApplication.getInstance();
        String msg = application.getString(msgId);
        showToast(msg);
    }

}

File: application/src/main/java/org/sheedon/common/handler/ToastHandler.kt
package org.sheedon.common.handler

import android.os.Looper
import androidx.annotation.StringRes
import org.sheedon.common.app.BaseApplication
import org.sheedon.common.utils.ToastUtils
import java.lang.Exception

/**
 * 消息提示者
 * @Author: sheedon
 * @Email: sheedonsun@163.com
 * @Date: 2022/1/5 6:15 下午
 */
class ToastHandler private constructor() {

    // ui 处理器
    private val uiHandler = BaseApplication.getUiHandler()

    companion object {
        private val INSTANCE = ToastHandler()

        @JvmStatic
        fun showToast(@StringRes res: Int) {
            try {
                val msg = ResConvertHandler.getInstance().convertString(res)
                showToast(msg)
            } catch (e: Exception) {
                e.printStackTrace()
            }
        }


        @JvmStatic
        fun showToast(msg: String?) {
            if (msg.isNullOrEmpty()) {
                return
            }
            if (Looper.myLooper() != INSTANCE.uiHandler.looper) {
                INSTANCE.uiHandler.post {
                    ToastUtils.showToast(BaseApplication.getInstance(), msg)
                }
            } else {
                ToastUtils.showToast(BaseApplication.getInstance(), msg)
            }
        }
    }

}